lib/logstash/outputs/kusto/ingestor.rb

```diff
@@ -20,15 +20,35 @@ class Ingestor
     LOW_QUEUE_LENGTH = 3
     FIELD_REF = /%\{[^}]+\}/
 
-    def initialize(ingest_url, app_id, app_key, app_tenant, database, table, json_mapping, delete_local, proxy_host , proxy_port , proxy_protocol,logger, threadpool = DEFAULT_THREADPOOL)
+    def initialize(ingest_url, app_id, app_key, app_tenant, managed_identity_id, database, table, json_mapping, delete_local, proxy_host , proxy_port , proxy_protocol,logger, threadpool = DEFAULT_THREADPOOL)
       @workers_pool = threadpool
       @logger = logger
-      validate_config(database, table, json_mapping,proxy_protocol)
+      validate_config(database, table, json_mapping,proxy_protocol,app_id, app_key, managed_identity_id)
       @logger.info('Preparing Kusto resources.')
 
       kusto_java = Java::com.microsoft.azure.kusto
       apache_http = Java::org.apache.http
-      kusto_connection_string = kusto_java.data.auth.ConnectionStringBuilder.createWithAadApplicationCredentials(ingest_url, app_id, app_key.value, app_tenant)
+      # kusto_connection_string = kusto_java.data.auth.ConnectionStringBuilder.createWithAadApplicationCredentials(ingest_url, app_id, app_key.value, app_tenant)
+      # If there is managed identity, use it. This means the AppId and AppKey are empty/nil
+      is_managed_identity = (app_id.nil? && app_key.empty?)
+      # If it is system managed identity, propagate the system identity
+      is_system_assigned_managed_identity = is_managed_identity && 0 == "system".casecmp(managed_identity_id)
+      # Is it direct connection
+      is_direct_conn = (proxy_host.nil? || proxy_host.empty?)
+      # Create a connection string
+      kusto_connection_string = begin
```

Comment:
do you need the begin end block?

Reply:
Got this fixed. This seems to be not necessary